sklearn/utils/_array_api.py

```diff
@@ -357,6 +367,28 @@ def _expit(X):
     return 1.0 / (1.0 + xp.exp(-X))
 
 
+def _nanmin(X, axis=None):
+    xp, _ = get_namespace(X)
+    if _is_numpy_namespace(xp):
+        return xp.asarray(numpy.nanmin(X, axis=axis))
+
+    else:
+        X = xp.min(xp.where(~xp.isnan(X), X, xp.asarray(+xp.inf)), axis=axis)
+        # Replace Infs from all NaN slices with NaN again
+        return xp.where(~xp.isinf(X), X, xp.asarray(xp.nan))
+
+
+def _nanmax(X, axis=None):
+    xp, _ = get_namespace(X)
+    if _is_numpy_namespace(xp):
+        return xp.asarray(numpy.nanmax(X, axis=axis))
+
+    else:
+        X = xp.max(xp.where(~xp.isnan(X), X, xp.asarray(-xp.inf)), axis=axis)
+        # Replace Infs from all NaN slices with NaN again
+        return xp.where(~xp.isneginf(X), X, xp.asarray(xp.nan))
```

Comment:
`isneginf` is not a part of the Array API spec and will fail with `cupy.array_api`. `xp.isneginf` will also fail for Array libraries that strictly follow the spec.